main.cpp

```diff
@@ -61,124 +69,411 @@ New/This/Pointers/References conclusion
+    DoubleType& add ( const FloatType& );
+    DoubleType& subtract ( const FloatType& );
+    DoubleType& multiply ( const FloatType& );
+    DoubleType& divide ( const FloatType& );
+    DoubleType& add ( const IntType& );
+    DoubleType& subtract ( const IntType& );
+    DoubleType& multiply ( const IntType& );
+    DoubleType& divide ( const IntType& );
 };
 
-double DoubleType::add (double x, double y)
+DoubleType& DoubleType::add ( const double y)
 {
-    return (x+y);
+    *pointerToDoubleValue += y;
+    return *this;
 }
 
-double DoubleType::subtract (double x, double y) 
+DoubleType& DoubleType::subtract ( const double y) 
 { 
-    return (x-y); 
+    *pointerToDoubleValue -= y;
+    return *this; 
 }
 
-double DoubleType::multiply (double x, double y) 
+DoubleType& DoubleType::multiply ( const double y) 
 { 
-    return (x*y); 
+    *pointerToDoubleValue *= y;
+    return *this; 
 }
 
-double DoubleType::divide (double x, double y) 
+DoubleType& DoubleType::divide ( const double y) 
 { 
-    if (abs(y) > 0) return (x/y);    
+    if (abs(y) > 0) *pointerToDoubleValue /= y;    
     
     std::cout << "Divide-by-zero warning  " << std::endl;
-    return (x/y);      
+    *pointerToDoubleValue /= 0;
+    return *this;      
 }
 
 struct IntType
 {
-    int add (int, int);
-    int subtract ( int, int);
-    int multiply ( int, int);
-    int divide (int, int);
+    int* pointerToIntValue = nullptr;
+
+     // default constructor assigns heap primitive if no value sent in
+    IntType() {
+        pointerToIntValue = new int();
+    }
+   
+    IntType( int intIn )
+    {
+        pointerToIntValue = new int ( intIn );
+    }
+    ~IntType()
+    {
+        delete pointerToIntValue;
+        pointerToIntValue = nullptr;
+    }
+
+    IntType& add ( int );
+    IntType& subtract ( int );
+    IntType& multiply ( int );
+    IntType& divide ( int );
+    IntType& divide ( const IntType& );
+    IntType& add ( const FloatType& );
+    IntType& subtract ( const FloatType& );
+    IntType& multiply ( const FloatType& );
+    IntType& divide ( const FloatType& );
+    IntType& add ( const DoubleType& );
+    IntType& subtract ( const DoubleType& );
+    IntType& multiply ( const DoubleType& );
+    IntType& divide ( const DoubleType& );
 };
 
-int IntType::add (int x, int y)
+IntType& IntType::add ( const int y )
 {
-    return (x+y);
+    *pointerToIntValue += y;
+    return *this;
+}
+
+IntType& IntType::subtract ( const int y ) 
+{ 
+    *pointerToIntValue -= y;
+    return *this;
 }
 
-int IntType::subtract (int x, int y) 
+IntType& IntType::multiply ( const int y ) 
 { 
-    return (x-y); 
+    *pointerToIntValue *= y;
+    return *this; 
 }
 
-int IntType::multiply (int x, int y) 
+IntType& IntType::divide ( const int y ) 
 { 
-    return (x*y); 
+    if (y != 0 ) 
+    {
+        *pointerToIntValue /=y; 
+        return *this;
+    } 
+    std::cout << "Cannot Divide Int by Zero! " << std::endl;
+    *pointerToIntValue = 0;
```

Comment:
Why are you changing the value?   You should just warn the user.

- [ ] remove assignment of 0